# Disadvantages of KNN Algorithm

The algorithm gets significantly slower as the number of examples and/or predictors/independent variables increase.

In [7]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
from sklearn import preprocessing
import time


df = pd.read_csv(r"C:\Users\user\Desktop\DATA SCIENCE\SpaceMissions (1).csv")

df['Mission Status New'] = LabelEncoder().fit_transform(df['Mission Status'])

data = df.drop(['Company', 'Launch Date', 'Launch Time', 'Launch Site','Vehicle Type','Payload Name','Payload Type','Payload Orbit','Failure Reason','Payload Mass (kg)','Mission Status'], axis = 1)

data = data.dropna()

start = time.time()

X_train, X_test, y_train, y_test = train_test_split(data.drop(['Mission Status New'],axis=1), 
                                                    data['Mission Status New'], test_size=0.20, 
                                                    random_state=10)
knnmodel = KNeighborsClassifier()
knnmodel.fit(X_train,y_train)

knnmodel_predictions = knnmodel.predict(X_test)

end = time.time()

accuracy_knnmodel = knnmodel.score(X_test,y_test)
print(accuracy_knnmodel*100,'%')

print("The time of execution of above program is :", end-start)

89.28571428571429 %
The time of execution of above program is : 0.014963150024414062


In [10]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
from sklearn import preprocessing
import time

data1 = pd.read_csv(r"C:\Users\user\Desktop\DATA SCIENCE\BA.csv")

start = time.time()

X_train, X_test, y_train, y_test = train_test_split(data1.drop(['Surname','CustomerId','RowNumber','Exited','Geography','Gender'],axis=1), 
                                                    data1['Exited'], test_size=0.20, 
                                                    random_state=20)
knnmodel = KNeighborsClassifier()
knnmodel.fit(X_train,y_train)

knnmodel_predictions = knnmodel.predict(X_test)

end = time.time()

accuracy_knnmodel = knnmodel.score(X_test,y_test)
print(accuracy_knnmodel*100,'%')

print("The time of execution of above program is :", end-start)

100.0 %
The time of execution of above program is : 3.635221242904663


In [8]:
df.shape

(150, 19)

The time of execution of above program is : 0.014963150024414062

In [11]:
data1.shape

(120000, 14)

The time of execution of above program is : 3.635221242904663